test/e2e/prometheus_test.go

```diff
@@ -5169,6 +5169,77 @@ func testPrometheusStatusScale(t *testing.T) {
 	}
 }
 
+// TODO: Also make sure no governing service was created?
+// What do we want to do? we want to make sure that when you are creating your own service and
+// the service is actually selected. But this is pretty self explanatory.
+// Something that we also want to check for is to make sure that if the service does not select
+// our prometheus, and the prometheus selects our service then we reject such a config.
+func testPrometheusServiceName(t *testing.T) {
+	t.Parallel()
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+	ns := framework.CreateNamespace(context.Background(), t, testCtx)
+	name := "test-servicename"
+
+	svc := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-service", name),
+			Namespace: ns,
+		},
+		Spec: v1.ServiceSpec{
+			Type: v1.ServiceTypeLoadBalancer,
+			Ports: []v1.ServicePort{
+				{
+					Name: "web",
+					Port: 9090,
+				},
+			},
+			Selector: map[string]string{
+				"prometheus":                   name,
+				"app.kubernetes.io/name":       "prometheus",
+				"app.kubernetes.io/instance":   name,
+				"app.kubernetes.io/managed-by": "prometheus-operator",
+			},
+		},
+	}
+
+	if _, err := framework.CreateOrUpdateService(context.Background(), ns, svc); err != nil {
+		t.Fatal(fmt.Errorf("failed to create app service: %w", err))
+	}
+
+	pm := framework.MakeBasicPodMonitor(name)
+	pm.Spec.Selector.MatchLabels = map[string]string{"prometheus": name}
+	pm.Spec.PodMetricsEndpoints = []monitoringv1.PodMetricsEndpoint{{Interval: "1s", Port: "web"}}
+
+	if _, err := framework.MonClientV1.PodMonitors(ns).Create(context.Background(), pm, metav1.CreateOptions{}); err != nil {
+		t.Fatal("failed to create PodMonitor: ", err)
+	}
+
+	framework.SetupPrometheusRBAC(context.Background(), t, testCtx, ns)
+
+	p := framework.MakeBasicPrometheus(ns, name, name, 1)
+	p.Spec.ServiceName = ptr.To(fmt.Sprintf("%s-service", name))
+
+	p, err := framework.CreatePrometheus(context.Background(), ns, p)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// TODO: Maybe we can just send a http request to the Prometheus pod to see if it is active?
```

Comment:
IIUC the existing test is enough: WaitForHealthyTargets() will target the custom service so if it replies ok, you know that everything is wired properly.

Reply:
I thought perhaps we could do away with the dummy target creation, but since it does mimic real life scenarios I think it should be good.